bookkeeper-server/src/main/java/org/apache/bookkeeper/client/TryPendingReadLacOp.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.client;
+
+import io.netty.buffer.ByteBuf;
+import java.util.List;
+
+import org.apache.bookkeeper.client.BKException.BKDigestMatchException;
+import org.apache.bookkeeper.net.BookieSocketAddress;
+import org.apache.bookkeeper.proto.BookieClient;
+import org.apache.bookkeeper.proto.BookkeeperInternalCallbacks.ReadLacCallback;
+import org.apache.bookkeeper.proto.checksum.DigestManager.RecoveryData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This op is try to read last confirmed without involving quorum coverage checking.
+ * Use {@link PendingLacOp} if you need quorum coverage checking.
```

Comment:
You mean PendingReadLacOp?

Reply:
I am not sure if we ever need quorum for LAC. LAC means it has quorum, so we can take greatest LAC.

Reply:
If we take out the coverageSet.checkCovered() check from PendingReadLacOp, then we don't need this right? I don't know why we have the quorum check in there, What scenario needs LAC  to be quorum checked? @sijie ?

Reply:
Haven't looked at this patch, but this coverage check is there because fencing is piggybacked on the LAC read.

Reply:
@jvrao @ivankelly  @sijie 
This feature has the same semantics as TryReadLastConfirmedOp:

TryPendingReadLacOp -> PendingReadLacOp
TryReadLastConfirmedOp -> ReadLastConfirmedOp

We are trying to advance LAC and return as soon as we have an useful value, without full coverage checks.

AFAIK This feature is a used for Long-Poll reads, which I am not going to implement soon, at least not within 4.9 timeline.
In Long Poll reads tests I have found this API *asyncTryReadLastConfirmed*  and _for consistency_  I have created a similar one for ExplicitLAC based reads.

Reply:
> without full coverage checks.

This is what I am trying to understand. Once we have LAC updated (either piggy-back or explicit) it means that it has met Write Quorum. So I don't understand what is the additional coverage check needed. This is not a fencing read or reading beyond LAC, this operation reads LAC which must have been updated by the writer after write quorum, hence it is persisted.